mm/Kconfig

```diff
@@ -3,6 +3,23 @@
 # see the file kconfig-language.txt in the NuttX tools repository.
 #
 
+choice
+	prompt "Build heap manager"
+	default MM_DEFAULT_MANAGER
+
+config MM_DEFAULT_MANAGER
+	bool "Default heap manager"
+	---help---
+		NuttX original memory manager strategy.
+
+config MM_CUSTOMIZE_MANAGER
```

Comment:
How are you including customised memory managers into the build?

Reply:
hi:
please check my patch "mm: add heap policy and anta logic to support mm debug on sim platform #3093",
this is my customize mm policy used under sim platform for sanitizers debug tool.

Reply:
There are many possibility here:

1. Integrate other heap manager policy(e.g. https://github.com/mattconte/tlsf)
2. Integrate the debug heap manger(e.g. https://github.com/j256/dmalloc)

ASAN(#3093) support on sim platform is just a demo but very useful to debug the tough memory problem with the advanced tool impossible on the real device.

Reply:
I was expecting something that can allow integration of other managers, like tlsf, through some build tricks, like the one we do with `context`.  #3093 seems to fix only one other option.

Reply:
Actually, we already done integration tlsf after this patch(user can select the builtin or tlsf through Kconfig), we initially has no plan to upstream this work because it depends on the 3rd party library, but if the community want a try, we can upstream our work.

Reply:
tlsf is BSD, I don't think this causes any issue, at least licensing wise. 
Have you seen any significant improvements when using tlsf? (I think this is a general discussion that we should move out of this PR)

Reply:
> Actually, we already done integration tlsf after this patch(user can select the builtin or tlsf through Kconfig), we initially has no plan to upstream this work because it depends on the 3rd party library, but if the community want a try, we can upstream our work.

If the license isn't compatible, maybe a Kconfig option to allow an "external" memory manager. Then the user can specify any manager, even their own custom one, as long as it provides the interfaces NuttX needs.

Reply:
How about we merge this PR first? @Ouss4 and @hartmannathan .

Reply:
@xiaoxiang781216 fine with me.